api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -245,6 +245,18 @@ public void getProgramByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
 		Assert.assertNull(Context.getProgramWorkflowService().getProgramByUuid("some invalid uuid"));
 	}
 	
+	/**
+	 * @see ProgramWorkflowService#getProgramByName(String)
+	 */
+	@Test
+	public void getProgramme_shouldGetProgrammeByName() {
```

Comment:
have you looked at the test coverage?

`getProgramByName` is fully tested, so please remove this test.

see current coverage of the class you are testing
https://coveralls.io/builds/14906713/source?filename=api%2Fsrc%2Fmain%2Fjava%2Forg%2Fopenmrs%2Fapi%2Fimpl%2FProgramWorkflowServiceImpl.java#L132

as the issue you are working on describes you should
* add unit tests (which require no database) into `ProgramWorkflowServiceUnitTest`
* tests that require a database into `ProgramWorkflowServiceTest`

we should try to add more tests without using a DB (we use an in-memory DB called h2) especially if they are so trivial like if I pass in null I expect such and such to happen. Please take the time to understand how the few tests in `ProgramWorkflowServiceUnitTest` work. We already have tons of tests which almost all run using a DB which in total takes quite some time. If we manage to write more tests without we will be able to reduce the time our builds run without loosing the security net that tests provide. Tests with DB are valuable but we should only have a selected few per service I think.

Reply:
@The getProgramByName should be changed to getProgram. Thanks

Reply:
@teleivo Shall I go on and create a new Exception class to handle concept being null ?